### PR TITLE
Rename recorder domain to hidden domain

### PR DESCRIPTION
**Pull request by M4GNV5**
This is bacially the leftovers of #1737.
Since the addition of `transcriber.yaml` by @aaronkvanmeerten the main functionality from the original PR already exist.
This PR simply renames the "recorder" domain to a more generic "hidden" domain, since it is used not only by the recorder, but also by the transcriber.

### CODE REVIEW

File: prosody/rootfs/defaults/conf.d/visitors.cfg.lua [hunk 1]
```diff
@@ -187,18 +187,23 @@ Component '{{ $VISITORS_MUC_PREFIX }}.v{{ $VISITOR_INDEX }}.{{ $VISITORS_XMPP_DO
...
-	-- The time in seconds, after which the limit for an IP address is lifted.
-	rate_limit_timeout = {{ $RATE_LIMIT_TIMEOUT }};
-	-- List of regular expressions for IP addresses that are not limited by this module.
-	rate_limit_whitelist = {
-      "127.0.0.1";
-      {{ range $index, $cidr := (splitList "," $RATE_LIMIT_ALLOW_RANGES | compact) -}}
-      "{{ $cidr }}";
-      {{ end -}}
-    };
+    rate_limit_login_rate = {{ $RATE_LIMIT_LOGIN_RATE }};
```
Comment (saghul) [inline prosody/rootfs/defaults/conf.d/visitors.cfg.lua:190]: Can you please avoid the unnecessary formatting changes here?

**Review by saghul — APPROVED** [review]
LGTM with a minor nit. @aaronkvanmeerten PTAL too!

File: jicofo/rootfs/defaults/jicofo.conf [hunk 2]
```diff
@@ -6,6 +6,7 @@
 {{ $JICOFO_AUTH_LIFETIME := .Env.JICOFO_AUTH_LIFETIME | default "24 hours" -}}
 {{ $ENABLE_SCTP := .Env.ENABLE_SCTP | default "1" | toBool -}}
 {{ $ENABLE_RECORDING := .Env.ENABLE_RECORDING | default "0" | toBool -}}
+{{ $ENABLE_TRANSCRIPTIONS := .Env.ENABLE_TRANSCRIPTIONS | default "0" | toBool -}}
```
Comment (aaronkvanmeerten) [inline jicofo/rootfs/defaults/jicofo.conf:9]: I believe this is being moved for alphabetizing (which I approve of), but also duplicates an existing entry for this variable further down in the file.  I suggest remove line 16 below if we're adding this one here.

**Review by aaronkvanmeerten — APPROVED** [review]
I approve, excepting the duplicated line I referenced i my comments should be removed.

File: prosody/rootfs/defaults/conf.d/visitors.cfg.lua (hunk 1 above)
Comment (M4GNV5) [reply prosody/rootfs/defaults/conf.d/visitors.cfg.lua:190]: I do agree this is mostly formatting, but before this PR there was mixed indentation (tabs/spaces) in this file.
For next time I assume the better approach would be to create a seperate PR for this?

Comment (saghul) [reply prosody/rootfs/defaults/conf.d/visitors.cfg.lua:190]: Generally speaking, yes. But since you are fixing spacing, let's leave it!

**Comment by M4GNV5** [comment]
I rebased this PR and added the one remaining change request. I hope this makes everything ready to be merged

**Review by saghul — APPROVED** [review]
